frontend/src/util/dom.ts

```diff
@@ -3,6 +3,44 @@ import { deepMap, onlyText } from "react-children-utilities";
 import { debounce } from "lodash";
 import { sleep } from "@/util/misc";
 
+/** get document root styles */
+const getRootStyles = () => window.getComputedStyle(document.documentElement);
```

Comment:
Various refactoring of these util funcs to be more modular and sensical. Delete ones that are no longer needed.